nntrainer/src/databuffer.cpp

```diff
@@ -583,34 +645,65 @@ int DataBuffer::setFeatureSize(unsigned int size) {
   int status = ML_ERROR_NONE;
   if (size == 0) {
     ml_loge("Error: batch size must be greater than 0");
-    for (int i = 0; i < NBUFTYPE; ++i)
-      validation[i] = false;
+    SET_VALIDATION(false);
     return ML_ERROR_INVALID_PARAMETER;
   }
 
   input_size = size;
+  long file_size = getFileSize(std::ref(train_stream));
+
   max_train = static_cast<unsigned int>(
-    getFileSize(train_file) /
-    (class_num * sizeof(int) + input_size * sizeof(float)));
+    file_size / (class_num * sizeof(int) + input_size * sizeof(float)));
   if (max_train < mini_batch) {
     ml_logw("Warning: number of training data is smaller than mini batch size");
   }
 
+  file_size = getFileSize(std::ref(val_stream));
   max_val = static_cast<unsigned int>(
-    getFileSize(val_file) /
-    (class_num * sizeof(int) + input_size * sizeof(float)));
+    file_size / (class_num * sizeof(int) + input_size * sizeof(float)));
   if (max_val < mini_batch) {
-    ml_logw("Warning: number of training data is smaller than mini batch size");
+    ml_logw("Warning: number of val data is smaller than mini batch size");
   }
 
+  file_size = getFileSize(std::ref(test_stream));
   max_test = static_cast<unsigned int>(
-    getFileSize(test_file) /
-    (class_num * sizeof(int) + input_size * sizeof(float)));
+    file_size / (class_num * sizeof(int) + input_size * sizeof(float)));
   if (max_test < mini_batch) {
-    ml_logw("Warning: number of training data is smaller than mini batch size");
+    ml_logw("Warning: number of test data is smaller than mini batch size");
   }
 
   return status;
 }
 
+void DataBuffer::displayProgress(const int count, BufferType type, float loss) {
+  int barWidth = 20;
+  float max_size = max_train;
+  switch (type) {
+  case BUF_TRAIN:
+    max_size = max_train;
+    break;
+  case BUF_VAL:
+    max_size = max_val;
+    break;
+  case BUF_TEST:
+    max_size = max_test;
+    break;
+  default:
+    break;
+  }
+  float progress = (((float)(count * mini_batch)) / max_size);
```

Comment:
progress can be over 1 if max_size < mini_batch

Reply:
This going to be updated incoming PR. Thanks.